flytekit/extend/backend/base_sensor.py

```diff
@@ -0,0 +1,47 @@
+from typing import Optional
+
+import cloudpickle
+import grpc
+from flyteidl.admin.agent_pb2 import (
+    RUNNING,
+    SUCCEEDED,
+    CreateTaskResponse,
+    DeleteTaskResponse,
+    GetTaskResponse,
+    Resource,
+)
+
+from flytekit.core.sensor_task import FileSensorConfig
+from flytekit.extend.backend.base_agent import AgentBase
+from flytekit.models.literals import LiteralMap
+from flytekit.models.task import TaskTemplate
+
+
+class SensorBase(AgentBase):
+    def __init__(self, task_type: str):
+        super().__init__(task_type=task_type, asynchronous=True)
+
+    async def poke(self, path: str) -> bool:
```

Comment:
this is not right? It should get some other information? This is ok for files only. What if I want to sense HivePartitions?

I think the api should be

```python
DerivedSensor(SensorBase[T]):
async def poke(self, v: T):
    ...
async def extract(self, context: grpc.ServicerContext,
        output_prefix: str,
        task_template: TaskTemplate,
        inputs: Optional[LiteralMap] = None) -> T:
    ....
```

Reply:
Happy to discuss

Reply:
done